File: TesteAndroidSantander/app/src/main/java/com/UI/MoreInfoAdapter.java
package com.UI;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.cerqueira.mellina.testeandroidsantander.R;

import java.util.List;

public class MoreInfoAdapter extends RecyclerView.Adapter<MoreInfoAdapter.ItemHolder>  {

    private List<MoreInfo> moreInfos;
    private Context context;

    public MoreInfoAdapter(Context context, List<MoreInfo> moreInfos) {
        this.context = context;
            this.moreInfos = moreInfos;

    }

    @Override
    public ItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.adapter_more_info, parent, false);
        ItemHolder holder = new ItemHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(ItemHolder holder, int position) {
        MoreInfo moreInfo = moreInfos.get(position);
        holder.txtPeriod.setText(moreInfo.getPeriod());
        holder.txtFund.setText(String.valueOf(moreInfo.getFund()));
        holder.txtCDI.setText(String.valueOf(moreInfo.getCDI()));

    }

    @Override
    public int getItemCount() {
        return moreInfos.size();
    }

    public class ItemHolder extends RecyclerView.ViewHolder {
        public TextView txtPeriod;
        public TextView txtFund;
        public TextView txtCDI;


        public ItemHolder(View view) {
            super(view);
            txtPeriod = view.findViewById(R.id.txtPeriod);
            txtFund = view.findViewById(R.id.txtFund);
            txtCDI = view.findViewById(R.id.txtCDI);

        }

    }
}

File: app/src/main/java/com/fuinha11/test/testeandroid/ui/adapter/MoreInfoAdapter.kt
package com.fuinha11.test.testeandroid.ui.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.fuinha11.test.testeandroid.R
import com.fuinha11.test.testeandroid.data.model.Investment
import kotlinx.android.synthetic.main.investment_info_cell.view.*

class MoreInfoAdapter(private val items : List<Investment.Info>) : RecyclerView.Adapter<MoreInfoAdapter.ViewHolder>() {
    override fun getItemCount(): Int {
        return items.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(items[position])
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.investment_info_cell, parent, false)
        return ViewHolder(view)
    }

    class ViewHolder (itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(item: Investment.Info) = with(itemView) {
            with(item){
                title.text = name
                if (data == null) {content.text = "Baixar"
                    download_arrow.visibility = android.view.View.VISIBLE
                    content.setTextColor(android.support.v4.content.ContextCompat.getColor(itemView.context, R.color.colorPrimaryDark))
                } else
                    content.text = data

            }
        }
    }
}